app/src/test/java/tech/ula/utils/ValidationUtilityTest.kt

```diff
@@ -24,206 +23,129 @@ class ValidationUtilityTest {
         assertFalse(credential.credentialIsValid)
-        assertEquals(credential.errorMessage, "Characters in username must be 1 to 30 characters with lowercase letters and/or numbers.")
+        assertEquals(credential.errorMessageId, R.string.error_username_invalid_characters)
     }
 
     @Test
-    fun `Username starts with underscore`() {
+    fun `Validation succeeds appropriately if username starts with underscore`() {
         val username = "_123abc"
         credential = validationUtility.validateUsername(username)
         assertTrue(credential.credentialIsValid)
-        assertEquals(credential.errorMessage, "")
+        assertEquals(credential.errorMessageId, 0)
+    }
+
+    @Test
+    fun `Validation fails appropriately if username has space`() {
+        val username = "user name"
+        credential = validationUtility.validateUsername(username)
+        assertFalse(credential.credentialIsValid)
+        assertEquals(credential.errorMessageId, R.string.error_username_invalid_characters)
+    }
+
+    @Test
+    fun `Validation fails appropriately if username is too long`() {
+        val username = "abcdefghijklmnopqrstuvwxyz123456"
+        credential = validationUtility.validateUsername(username)
+        assertFalse(credential.credentialIsValid)
+        assertEquals(credential.errorMessageId, R.string.error_username_invalid_characters)
     }
 
     @Test
-    fun `Username is in blacklist`() {
+    fun `Validation fails appropriately if username is in blacklist`() {
         val username = "root"
-        whenever(resources.getString(R.string.error_username_in_blacklist))
-                .thenReturn("The username: `root` is reserved and cannot be used as a filesystem username.")
 
         credential = validationUtility.validateUsername(username)
         assertFalse(credential.credentialIsValid)
-        assertEquals(credential.errorMessage, "The username: `root` is reserved and cannot be used as a filesystem username.")
+        assertEquals(credential.errorMessageId, R.string.error_username_in_blacklist)
     }
 
     @Test
-    fun `password is empty`() {
+    fun `Validation fails appropriately if password is empty`() {
         val password = ""
-        val vncPassword = defaultPassword
 
-        credential = validationUtility.validatePasswords(password, vncPassword)
+        credential = validationUtility.validatePassword(password)
         assertFalse(credential.credentialIsValid)
-        assertEquals(credential.errorMessage, "Each field must be entered!")
+        assertEquals(credential.errorMessageId, R.string.error_empty_field)
     }
 
     @Test
-    fun `vnc password is empty`() {
-        val password = defaultPassword
+    fun `Validation fails appropriately if vnc password is empty`() {
         val vncPassword = ""
 
-        credential = validationUtility.validatePasswords(password, vncPassword)
+        credential = validationUtility.validatePassword(vncPassword)
```

Comment:
You call `validatePassword` instead of `validateVncPassword` here.

Reply:
Good catch!